test/cri-containerd/container_test.go

```diff
@@ -309,6 +310,222 @@ func Test_RunContainer_VirtualDevice_GPU_LCOW(t *testing.T) {
+				"io.microsoft.virtualmachine.lcow.kerneldirectboot":                  "false",
+				"io.microsoft.virtualmachine.computetopology.memory.allowovercommit": "false",
+				"io.microsoft.virtualmachine.lcow.preferredrootfstype":               "initrd",
+				"io.microsoft.virtualmachine.devices.virtualpmem.maximumcount":       "0",
+				"io.microsoft.virtualmachine.lcow.vpcienabled":                       "true",
+				// we believe this is a sufficiently large high MMIO space amount for this test.
+				// if a given gpu device needs more, this test will fail to create the container
+				// and may hang.
+				"io.microsoft.virtualmachine.computetopology.memory.highmmiogapinmb": "64000",
+				"io.microsoft.virtualmachine.lcow.bootfilesrootpath":                 testGPUBootFiles,
+			},
+		},
+		RuntimeHandler: lcowRuntimeHandler,
+	}
+
+	podID := runPodSandbox(t, client, podctx, sandboxRequest)
+	defer removePodSandbox(t, client, podctx, podID)
+	defer stopPodSandbox(t, client, podctx, podID)
+
+	device := &runtime.Device{
+		HostPath: "gpu://" + testDeviceInstanceID,
+	}
+
+	containerGPURequest := &runtime.CreateContainerRequest{
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: "GPU-Container",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageLcowAlpine,
+			},
+			Command: []string{
+				"top",
+			},
+			Devices: []*runtime.Device{
+				device,
+			},
+			Linux: &runtime.LinuxContainerConfig{},
+			Annotations: map[string]string{
+				"io.microsoft.container.gpu.capabilities": "utility",
+			},
+		},
+		PodSandboxId:  podID,
+		SandboxConfig: sandboxRequest.Config,
+	}
+
+	containerNoGPURequest := &runtime.CreateContainerRequest{
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: "No-GPU-Container",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageLcowAlpine,
+			},
+			Command: []string{
+				"top",
+			},
+			Linux: &runtime.LinuxContainerConfig{},
+		},
+		PodSandboxId:  podID,
+		SandboxConfig: sandboxRequest.Config,
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
+	// create container with a GPU present
+	gpuContainerID := createContainer(t, client, ctx, containerGPURequest)
+	defer removeContainer(t, client, ctx, gpuContainerID)
+	startContainer(t, client, ctx, gpuContainerID)
+	defer stopContainer(t, client, ctx, gpuContainerID)
+
+	noGPUContainerID := createContainer(t, client, ctx, containerNoGPURequest)
+	defer removeContainer(t, client, ctx, noGPUContainerID)
+	startContainer(t, client, ctx, noGPUContainerID)
+	defer stopContainer(t, client, ctx, noGPUContainerID)
+
+	// verify that we can't access the GPU in the No-GPU-Container
+	cmd := []string{"ls", "/dev/nvidia0"}
+	containerExecReq := &runtime.ExecSyncRequest{
+		ContainerId: noGPUContainerID,
+		Cmd:         cmd,
+		Timeout:     20,
+	}
+	response := execSync(t, client, ctx, containerExecReq)
+	if len(response.Stderr) == 0 {
```

Comment:
Same as above
```go
if response.ExitCode != 0 {
		t.Fatalf("failed with exit code %d running 'ls /dev/nvidia0': %s", response.ExitCode, string(response.Stderr))
}
```

Reply:
This one is intended to fail, and I want to specifically differentiate between stdout and stderr messages as the presence of either indicates a specific problem.

Reply:
Ok makes sense thanks for clarifying